test/blackbox-tests/test-cases/actions/cond.t

```diff
@@ -0,0 +1,134 @@
+Testing (case) action.
+
+  $ cat > dune-project << EOF
+  > (lang dune 3.11)
+  > EOF
+
+Basic usage:
+
+  $ cat > dune << EOF
+  > (rule
+  >  (alias foo)
+  >  (action
+  >   (cond
+  >    ((= 1 2) (echo " 1 is 2"))
+  >    ((= 2 3) (echo "2 is 3"))
+  >    ((< 3 4) (echo "3 is less than 4"))
+  >    (_ (echo "shouldn't happen")))))
+  > EOF
+
+  $ dune build @foo
+  3 is less than 4
+
+Missing default field:
+
+  $ cat > dune << EOF
+  > (rule
+  >  (alias foo)
+  >  (action
+  >   (cond
+  >    ((= 1 2) (echo " 1 is 2")))))
+  > EOF
+
+  $ dune build @foo
+  File "dune", line 5, characters 4-11:
+  5 |    ((= 1 2) (echo " 1 is 2")))))
+          ^^^^^^^
+  Error: Only the default case can be at the end.
+  Hint: Add a (_ (...)) case at the end.
+  [1]
+
+Two default fields:
+
+  $ cat > dune << EOF
+  > (rule
+  >  (alias foo)
+  >  (action
+  >   (case %{read-lines:version}
+  >    (_ (echo "version is 1"))
+  >    (2 (echo "version is 2"))
+  >    (_ (echo "version is 3")))))
+  > EOF
+
+  $ cat > dune << EOF
+  > (rule
+  >  (alias foo)
+  >  (action
+  >   (cond
+  >    (_ (echo " 1 is 2"))
+  >    ((= 2 3) (echo "2 is 3"))
+  >    (_ (echo "3 is less than 4")))))
+  > EOF
+
+  $ dune build @foo
+  Error: Multiple default cases.
+  File "dune", line 5, characters 4-5:
+  5 |    (_ (echo " 1 is 2"))
+          ^
+  
+  File "dune", line 7, characters 4-5:
+  7 |    (_ (echo "3 is less than 4")))))
+          ^
+  
+  [1]
+
+Last case is not default case:
+
+  $ cat > dune << EOF
+  > (rule
+  >  (alias foo)
+  >  (action
+  >   (cond
+  >    ((= 1 2) (echo " 1 is 2"))
+  >    ((= 2 3) (echo "2 is 3"))
+  >    (_ (echo "shouldn't happen"))
+  >    ((< 3 4) (echo "3 is less than 4")))))
+  > EOF
+
+  $ dune build @foo
+  File "dune", line 8, characters 4-11:
+  8 |    ((< 3 4) (echo "3 is less than 4")))))
+          ^^^^^^^
+  Error: Only the default case can be at the end.
```

Comment:
If I were consuming this error message I'd prefer to be told that the default case appears somewhere other than the end of the case statement in this instance.

Reply:
This is trickier since we parse the list backwards to check the (reversed) first is a default case. In order to find the default case we would have to parse the rest of them. There is also no guarantee that there is even a default case.